subsystem/metrics_test.go

```diff
@@ -588,6 +594,34 @@ var _ = Describe("Metrics tests", func() {
 			assertValidationEvent(ctx, day2ClusterID, "master-0", models.HostValidationIDAPIVipConnected, false)
 		})
 
+		It("'belongs-to-majority-group' failed", func() {
+
+			// create a validation success
+			h1 := registerNode(ctx, clusterID, "h1")
+			h2 := registerNode(ctx, clusterID, "h2")
+			h3 := registerNode(ctx, clusterID, "h3")
+			h4 := registerNode(ctx, clusterID, "h4")
+			generateFullMeshConnectivity(ctx, "1.2.3.10", h1, h2, h3, h4)
+			waitForHostValidationStatus(clusterID, *h1.ID, "success", models.HostValidationIDBelongsToMajorityGroup)
+
+			// create a validation failure
+			generateFullMeshConnectivity(ctx, "1.2.3.10", h2, h3, h4)
+			waitForHostValidationStatus(clusterID, *h1.ID, "failure", models.HostValidationIDBelongsToMajorityGroup)
+
+			// check generated events
+			assertValidationEvent(ctx, clusterID, "h1", models.HostValidationIDBelongsToMajorityGroup, true)
+
+			// check generated metrics
+			metricCounter := getValidationMetricCounter(clusterID, models.HostValidationIDBelongsToMajorityGroup, hostValidationChangedMetric)
+			// this specific case can create a short timeframe in which another host is failing on that validation and will
+			// be later fixed by the next refresh status cycle because generating a full mesh connectivity isn't an atomic
+			// action, therefore, in this test we will check that at least the expected failing host is failing but not fail
+			// the test if other hosts fails as well.
+			Expect(metricCounter >= 1).To(BeTrue())
+			metricsDeregisterCluster(ctx, clusterID)
+			assertValidationMetricCounter(clusterID, models.HostValidationIDBelongsToMajorityGroup, hostValidationFailedMetric, 1)
```

Comment:
There is a possibility that h2 will still be in failed state.  Is it OK for this assertion?